runtime/src/main/java/com/flipkart/flux/dao/StatesDAOImpl.java

```diff
@@ -131,6 +131,16 @@ public List findStatesByStatus(ShardId shardId, String stateMachineName, Timesta
         return query.list();
     }
 
-
-
+    @Override
+    @Transactional
+    @SelectDataSource(type = DataSourceType.READ_WRITE, storage = Storage.SHARDED)
+    public List findStatesByDependentEvent(String stateMachineId, String eventName) {
+        Query query;
+        String queryString = "select id, stateMachineId, status from State where stateMachineId = :stateMachineId" +
```

Comment:
format this in one single string

Reply:
It's formatted, please view it as a whole file.